ui/src/components/AgentsProvider.tsx

```diff
@@ -70,60 +72,44 @@ interface AgentsProviderProps {
 }
 
 export function AgentsProvider({ children }: AgentsProviderProps) {
-  const [agents, setAgents] = useState<AgentResponse[]>([]);
-  const [error, setError] = useState("");
-  const [loading, setLoading] = useState(true);
-  const [tools, setTools] = useState<ToolsResponse[]>([]);
-  const [models, setModels] = useState<ModelConfig[]>([]);
-
-  const fetchAgents = useCallback(async () => {
-    try {
-      setLoading(true);
-      const agentsResult = await getAgents();
-
-      if (!agentsResult.data || agentsResult.error) {
-        throw new Error(agentsResult.error || "Failed to fetch agents");
-      }
-
-      setAgents(agentsResult.data);
-      setError("");
-    } catch (err) {
-      setError(err instanceof Error ? err.message : "An unexpected error occurred");
-    } finally {
-      setLoading(false);
-    }
-  }, []);
-
-  const fetchModels = useCallback(async () => {
-    try {
-      const response = await getModelConfigs();
-      if (!response.data || response.error) {
-        throw new Error(response.error || "Failed to fetch models");
-      }
-
-      setModels(response.data);
-      setError("");
-    } catch (err) {
-      console.error("Error fetching models:", err);
-      setError(err instanceof Error ? err.message : "An unexpected error occurred");
-    } finally {
-      setLoading(false);
-    }
-  }, []);
-
-  const fetchTools = useCallback(async () => {
-    try {
-      setLoading(true);
-      const response = await getTools();
-      setTools(response);
-      setError("");
-    } catch (err) {
-      console.error("Error fetching tools:", err);
-      setError(err instanceof Error ? err.message : "An unexpected error occurred");
-    } finally {
-      setLoading(false);
-    }
-  }, []);
+  const { getSwrConfig } = useSettings();
+  
+  // Create typed fetchers
+  const agentsFetcher = createTypedFetcher<AgentResponse[]>();
+  const toolsFetcher = createTypedFetcher<ToolsResponse[]>();
+  const modelsFetcher = createTypedFetcher<ModelConfig[]>();
```

Comment:
instead of using a completely different fetcher -- can you use the existing fetch functions we have in the /actions foldeR? (e.g. agents.ts and the getAgents fetch function there -- that one explicitly sorts the results alphabetically, so we always get the same order; right now, every refresh is in a different order)